src/components/FilterDropdown/index.js

```diff
@@ -79,22 +94,23 @@ const FilterDropdown = ({
           >
             All
           </Dropdown.Item>
+          ) : ''}
           {dropdownItems?.map((item, idx) => {
             return (
               <Dropdown.Item
                 key={idx}
-                className={
-                  activeItem === item.name
-                    ? style.activeItem
-                    : style.dropdownItems
+                className={`${onHardCodeStyle}
+                ${activeItem === item.name
+                  ? style.activeItem
+                  : style.dropdownItems}`
                 }
                 onClick={() => {
                   setActiveItem(item.name);
-                  setFilter(item.name);
+                  {onDataNeeded ? setFilter(item.name) : onSetFilter(item.value);}
```

Comment:
can we return the whole object instead of the exact value if we pass onDataNeeded as true?
```suggestion
                  {onDataNeeded ? setFilter(item.name) : onSetFilter(item);}
```
then in the parent component, we retrieve the value only